diff --git a/drivers/pinctrl/intel/pinctrl-cherryview.c b/drivers/pinctrl/intel/pinctrl-cherryview.c
index c43b1e9..cb13338 100644
--- a/drivers/pinctrl/intel/pinctrl-cherryview.c
+++ b/drivers/pinctrl/intel/pinctrl-cherryview.c
@@ -492,7 +492,7 @@ static const struct chv_community north_community = {
 	 * 8 interrupts. The upper half (8-15) can only be used to trigger
 	 * GPEs.
 	 */
-	.nirqs = 8,
+	.nirqs = 9,
 };
 
 static const struct pinctrl_pin_desc east_pins[] = {
